### ISSUE TITLE
`cachix push` fails on paths which are `not valid, skipping`

### ISSUE THREAD

**Issue by srid** [cachix/cachix#529]
`cachix push` can, from time to time, fail with errors like this:

```
[93mWarning: /nix/store/8bw9z7ddxpgjmmjb3ryx1ac8rs7csjqf-nammayatri-exes is not valid, skipping[0m
cachix: FatalError {fatalErrorMessage = "no such path /nix/store/8bw9z7ddxpgjmmjb3ryx1ac8rs7csjqf-nammayatri-exes"}
```

Note that his path does exist actually.

FTR, here's what the CI does:

```sh
    env -i PATH="$PATH" nix develop --no-update-lock-file --profile dev-profile -c echo
    cachix push "''${CACHE}" dev-profile
```

https://github.com/juspay/jenkins-nix-ci/blob/e2089fd25872b33cafb710f1a1dff15fefab7140/nix/jenkins/features/cachix/cachixPush.nix#L27

---

Domen suggested to downgrade to 1.3.3 which shouldn't have this issue.

**Comment by colemickens** [comment]
What's the core issue? I'm seeing this today as well, and my remote builder that I push from isn't getting upgraded...

**Comment by domenkozar** [comment]
I suggest to rollback to 1.3.3 until I resolve this in a different way. I'll do that for nixpkgs.

**Comment by sandydoo** [comment]
> Given https://github.com/cachix/cachix/commit/a3d1a998254282e28e0cbdb3573af61d58b88e31 should I upgrade to cachix 1.5 and give it a try again to see if this issue is resolved?

@srid, yes, we've rolled back the store API changes. This should be resolved in `v1.5`.
